File: all-submissions/0015-3sum/solution.java
public class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        List<List<Integer>> answer = new ArrayList<>();
        Arrays.sort(nums);
        for(int i=0; i<nums.length-1;i++) {
            int j = i+1;
            int k = nums.length-1;
            while(j<k) {
                int tripletSum = nums[i] + nums[j] + nums[k];
                if (tripletSum == 0) {
                    answer.add(new ArrayList<>(Arrays.asList( nums[i],nums[j],nums[k])));
                    j++;
                    k--;
                    while(j<k && nums[j] == nums[j-1]) j++;
                    while(j<k && nums[k] == nums[k+1]) k--;
                } else {
                    if(tripletSum > 0){
                        k--;
                    } else {
                        j++;
                    }
                }
            }
            while (i + 1 < nums.length && nums[i + 1] == nums[i]) i++;
        }
        return answer;
        // Your code goes here
    }
}

File: all-submissions/0015-3sum/solution.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        if nums is None or len(nums) < 3: return []
        nums.sort()
        answer = []
        for i in range(len(nums)-2):
            if (i>0 and nums[i] == nums[i-1]): 
                continue
            j = i +1 
            k = len(nums)-1
            while(j < k):
                sum = nums[i] + nums[j] + nums[k] 
                if sum == 0:
                    answer.append([nums[i],nums[j],nums[k]])
                    j+=1
                    k-=1
                    while(0 < j < len(nums) and nums[j] == nums[j-1]): j+=1
                    while(k<len(nums)-1 and k>=0 and nums[k] == nums[k+1]): k -=1
                elif sum > 0:
                    k-=1
                else: 
                    j+=1
        return answer

